core/client/fs/src/main/java/alluxio/client/block/policy/CapacityBaseRandomPolicy.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.block.policy;
+
+import alluxio.client.block.BlockWorkerInfo;
+import alluxio.client.block.policy.options.GetWorkerOptions;
+import alluxio.wire.WorkerNetAddress;
+
+import java.util.TreeMap;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.atomic.AtomicLong;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Random load block base on capacity of worker,
+ * used for cluster of workers with different capacity.
+ * Then larger capacity worker will receive more load request.
+ *
+ */
+@ThreadSafe
+public class CapacityBaseRandomPolicy implements BlockLocationPolicy {
+
+  @Nullable
+  @Override
+  public WorkerNetAddress getWorker(GetWorkerOptions options) {
+    Iterable<BlockWorkerInfo> blockWorkerInfos = options.getBlockWorkerInfos();
+    //key is the floor of capacity range the worker cover
+    TreeMap<Long, BlockWorkerInfo> capacityFloorMap = new TreeMap<>();
+    AtomicLong totalCapacity = new AtomicLong(0L);
+    blockWorkerInfos.forEach(workerInfo -> {
+      if (workerInfo.getCapacityBytes() > 0) {
+        long capacityRangeFloor = totalCapacity.getAndAdd(workerInfo.getCapacityBytes());
+        capacityFloorMap.put(capacityRangeFloor, workerInfo);
```

Comment:
You may want to include this case in the test too, for example have multiple workers with same capacity and see what happens. This can actually be a very common case where majority of workers have the same capacity.

Reply:
> use the capacity as key, two workers with the same capacity (unlucky)

the key is the floor of the capacity range,  not its capacity.  the key is grow with getAndAdd method
eg. three workers with 10T capacity. the first key is 0 (range 0~10), the second key is 10 (range 10~20), the third key is 20 ( range 20~30)

Reply:
> You may want to include this case in the test too, for example have multiple workers with same capacity and see what happens. This can actually be a very common case where majority of workers have the same capacity.

@jiacheliu3 so, I think is not necessary to add more test for multiple workers with same capacity for my explanation above

Reply:
Good catch. I didn't notice the flooring.

I think it's still necessary to add some more tests for multiple workers with the same capacity. You can just duplicate your existing `getSuccessWorker()` and make them `getWorkerDifferentCapacity()` and `getWorkerSameCapacity()`.